publiccloud: Use VM id instead of name for Azure

cfconrad opened the pull request:
In azure a VM can be specified by resource group (rg) plus instance name
or by instance ID.
With this patch we move from rg+name to ID, which make later usage more
easy, but also breaks the implicit assumption that name and rg is always
equal.

- Related ticket: https://progress.opensuse.org/issues/66116
- Verification run: https://openqa.suse.de/t4173399

cfconrad commented:
@ldevulder I didn't found a solution without touching the terraform files. 
Is it possible to do something like https://github.com/SUSE/ha-sap-terraform-deployments/commit/527ec988e702e2dce2d17260454c5e53acff83da in your tf files? Of course for all types of VM's you are creating there.

I think this is the best option, because if I understand your tf files correct, then you are creating more then one resource group with on `terraform apply` right? because of this: https://github.com/SUSE/ha-sap-terraform-deployments/blob/master/azure/modules/hana_node/outputs.tf#L14

ldevulder commented:
I tested the PR on my lab and I had this issue: http://1b210.qa.suse.de/tests/6654#step/sles4sap/525.

The data I have for id is:

    "cluster_nodes_id": {
      "sensitive": false,
      "type": [
        "tuple",
        [
          "string",
          "string"
        ]
      ],
      "value": [
        "/subscriptions/c011786b-59d7-4817-880c-7cd8a6ca4b19/resourceGroups/rg-ha-sap-qashapopenqa/providers/Microsoft.Compute/virtualMachines/vmhana01",
        "/subscriptions/c011786b-59d7-4817-880c-7cd8a6ca4b19/resourceGroups/rg-ha-sap-qashapopenqa/providers/Microsoft.Compute/virtualMachines/vmhana02"
      ]
    }

cfconrad commented:
Sorry for being not clear here.
Your new output is called `cluster_nodes_id`, don't know if you also changed this https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/publiccloud/provider.pm#L395 line. It should use now `cluster_nodes_id`, if not - this might be the problem.

If you need the `resource group` or `vm name` seperated, I added that helper for it https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/10115/files#diff-1179ed55c04f7172e84ef29dffa9167fR299

In our general `azure.tf` file, I just changed the content of `vm_name` to display the `id` instead of `name` - sorry for confusion. I think it is worth to refactor it some time to `vm_ids` but I tried to keep the current change small.

cfconrad commented:
I will set this PR again to WIP, unless we solved this completely.

ldevulder commented:
> In our general `azure.tf` file, I just changed the content of `vm_name` to display the `id` instead of `name` - sorry for confusion. I think it is worth to refactor it some time to `vm_ids` but I tried to keep the current change small.

It's OK now: http://1b210.qa.suse.de/tests/6664
As we discussed I didn't change the `cluster_nodes_name` to `cluster_nodes_id`.

This PR can be merged for me!
